cmd/vdev_id/vdev_id

```diff
@@ -175,28 +238,40 @@ sas_handler() {
 		# we have to append the -part suffix directly in the
 		# helper.
 		if [ "$DEVTYPE" != "partition" ] ; then
-			PART=`echo $DM_NAME | awk -Fp '/p/{print "-part"$2}'`
+			PART=$(echo "$DM_NAME" | awk -Fp '/p/{print "-part"$2}')
 		fi
 
 		# Strip off partition information.
-		DM_NAME=`echo $DM_NAME | sed 's/p[0-9][0-9]*$//'`
+		DM_NAME=$(echo "$DM_NAME" | sed 's/p[0-9][0-9]*$//')
 		if [ -z "$DM_NAME" ] ; then
 			return
 		fi
 
-		# Get the raw scsi device name from multipath -ll. Strip off
-		# leading pipe symbols to make field numbering consistent.
-		DEV=`multipath -ll $DM_NAME |
-			awk '/running/{gsub("^[|]"," "); print $3 ; exit}'`
+		# Utilize DM device name to gather subordinate block devices
+		# using sysfs to avoid userspace utilities
+		DMDEV=$(echo "$DM_NAME" | awk '{gsub("../", " "); print $NF}')
+
+		# Use sysfs pointers in /sys/block/dm-X/slaves because using
+		# userspace tools creates lots of overhead and should be avoided
+		# whenever possible. Use awk to isolate lowest instance of
+		# sd device member in dm device group regardless of string
+		# length.
+		DEV=$(ls "/sys/block/$DMDEV/slaves" | awk '
+			{ len=sprintf ("%20s",length($0)); gsub(/ /,0,str); a[NR]=len "_" $0; }
+			END {
+				asort(a)
+				print substr(a[1],22)
+			}')
+
```

Comment:
I'm running into issues here.  So for my JBOD node, `DMDEV` is something like `35000c500a61d6aeb` (found by adding debug prints in your script), but what I think you want is for it to be like `dm-114`.  That way, the `/sys/block/$DMDEV` lookup works correctly.  I'm trying to work on a fix for it, but won't be able to work on it again until Jan 1st.

Reply:
Hi Tony - Thanks for point this out. We ourselves ran into this. We should be having the fix quickly. There are other minor modification/improvements being done as we see in our testing.

Reply:
Tony, this was missed in the latest push.

`DMDEV=$(ls -l --full-time /dev/mapper | grep $DM_NAME | awk '{gsub("../", " "); print $NF}')`

```
# ls -ali /dev/disk/by-vdev/ | grep AEON_TEST0
640522 lrwxrwxrwx. 1 root root  11 Jan 28 22:41 AEON_TEST0-1-0 -> ../../dm-21
642759 lrwxrwxrwx. 1 root root  11 Jan 28 22:41 AEON_TEST0-1-1 -> ../../dm-40
...
#
```

Reply:
Hi Tony - Please do try out the latest version. Please do let us know how does it go.

Reply:
Thanks, I'll give it a try today

Reply:
Works now!